base/sharedarray.jl

```diff
@@ -512,8 +512,6 @@ function copy!(S::SharedArray, R::SharedArray)
     return S
 end
 
-complex(S1::SharedArray,S2::SharedArray) = convert(SharedArray, complex(S1.s, S2.s))
```

Comment:
I would not recommend `broadcast(::typeof(complex), ...)`, because it will not be called in the presence of fusion etcetera.

If you want a `SharedArray` output, one option is to use in-place assignment (`mysharedarray .= complex.(a,b)`).   Another possibility would to define `Base.Broadcast.containertype` promotion methods to always produce a `SharedArray` from `broadcast` on `SharedArray` inputs, if that is the desired behavior.